pre_push.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+"""Run static analysis on the project."""
+
+import argparse
+import sys
+from os import path
+from subprocess import CalledProcessError, check_call
+
+current_directory = path.abspath(path.join(__file__, ".."))
+
+
+def do_process(args, shell=False):
+    """Run program provided by args.
+
+    Return True on success.
+
+    Output failed message on non-zero exit and return False.
+
+    Exit if command is not found.
+
+    """
+    print(f"Running: {' '.join(args)}")
+    try:
+        check_call(args, shell=shell)
```

Comment:
Wouldn't `check_output` make more sense as it actually waits for the response before returning control.

Reply:
Like I said above, this is copied from PRAW and I did not write the script. Feel free to open a PR to improve it, as I feel it is outside of this PR's scope.

Reply:
Sure. Sounds like a good idea.